apollo-federation/src/supergraph/mod.rs

```diff
@@ -83,6 +87,64 @@ use crate::schema::type_and_directive_specification::TypeAndDirectiveSpecificati
 use crate::schema::type_and_directive_specification::UnionTypeSpecification;
 use crate::utils::FallibleIterator;
 
+#[derive(Debug)]
+#[allow(unused)]
+pub struct Supergraph<S> {
+    pub state: S,
+}
+
+impl Supergraph<Merged> {
+    pub fn assume_valid(self) -> Supergraph<Satisfiable> {
+        todo!("unimplemented")
+    }
+}
+
+impl Supergraph<Satisfiable> {
+    /// Generates an API Schema from this supergraph schema. The API Schema represents the combined
+    /// API of the supergraph that's visible to end users.
+    pub fn to_api_schema(
+        &self,
+        options: ApiSchemaOptions,
+    ) -> Result<ValidFederationSchema, FederationError> {
+        api_schema::to_api_schema(self.state.schema.clone(), options)
+    }
+}
+
+#[derive(Clone, Debug)]
+#[allow(unused)]
+pub struct Merged {
```

Comment:
I liked the original `Initial` -> `Validated` states.... but since we define the same subgraph states then we would have to rename those whenever importing both states....